internal/ansible/ansible.go

```diff
@@ -20,131 +20,260 @@ import (
-	return func(ap *playbook.AnsiblePlaybookCmd) {
-		ap.StdoutCallback = stdoutCallback
+// withAnsibleVerbosity set the ansible-runner verbosity.
+func withAnsibleVerbosity(verbosity int) runnerOption {
+	return func(r *Runner) {
+		r.ansibleVerbosity = verbosity
 	}
 }
 
-// WithConnectionOptions defines the ansible's playbook connection options.
-func WithConnectionOptions(options *options.AnsibleConnectionOptions) PlaybookOption {
-	return func(ap *playbook.AnsiblePlaybookCmd) {
-		ap.ConnectionOptions = options
+type cmdFuncType func(verbosity int) *exec.Cmd
+
+// playbookCmdFunc mimics https://github.com/operator-framework/operator-sdk/blob/707240f006ecfc0bc86e5c21f6874d302992d598/internal/ansible/runner/runner.go#L75-L90
+func playbookCmdFunc(path string) (cmdFuncType, error) {
+	runnerBinary, err := runnerutil.RunnerBinary()
+	if err != nil {
+		return nil, err
 	}
-}
 
-// WithOptions defines the ansible's playbook options.
-func WithOptions(options *playbook.AnsiblePlaybookOptions) PlaybookOption {
-	return func(ap *playbook.AnsiblePlaybookCmd) {
-		ap.Options = options
+	wd, err := os.Getwd()
+	if err != nil {
+		return nil, err
 	}
+
+	return func(verbosity int) *exec.Cmd {
+		cmdArgs := []string{"run", wd}
+		cmdOptions := []string{
+			"-p", path,
+		}
+
+		// check the verbosity since the exec.Command will fail if an arg as "" or " " be informed
+		if verbosity > 0 {
+			cmdOptions = append(cmdOptions, runnerutil.AnsibleVerbosityString(verbosity))
+		}
+		// gosec is disabled here because of G204. We should pay attention that user can't
+		// make command injection via command argument
+		return exec.Command(runnerBinary, append(cmdArgs, cmdOptions...)...) //nolint:gosec
+	}, nil
 }
 
-// Init initializes pbCmd from parameters
-func (p Parameters) Init(ctx context.Context) (*PbCmd, error) {
+// roleCmdFunc mimics https://github.com/operator-framework/operator-sdk/blob/707240f006ecfc0bc86e5c21f6874d302992d598/internal/ansible/runner/runner.go#L92-L118
+func roleCmdFunc(path string) (cmdFuncType, error) {
+	rolePath, roleName := filepath.Split(path)
+	runnerBinary, err := runnerutil.RunnerBinary()
+	if err != nil {
+		return nil, err
+	}
 
-	// Read playbooks filename from dir
-	pbList, err := readDir(p.Dir, p.ExcludedFiles)
+	wd, err := os.Getwd()
 	if err != nil {
 		return nil, err
 	}
-	return NewAnsiblePlaybook(WithPlaybooks(pbList),
-		// `ansible-playbook` cmd output JSON Serialization
-		WithStdoutCallback("json"),
-		// e.g options.AnsibleConnectionOptions{Connection: "local"}
-		WithConnectionOptions(&options.AnsibleConnectionOptions{}),
-		// e.g playbook.AnsiblePlaybookOptions{Inventory: "127.0.0.1,"}
-		WithOptions(&playbook.AnsiblePlaybookOptions{})), nil
+
+	return func(verbosity int) *exec.Cmd {
+		// check the verbosity since the exec.Command will fail if an arg as "" or " " be informed
+
+		cmdOptions := []string{
+			"--role", roleName,
+			"--roles-path", rolePath,
+			"--hosts", "localhost",
```

Comment:
For the use of `localhost`, as I remember in Ansible Operator it is designed for manipulating k8s resources and it talks to a local proxy, so there's no need to support hosts w/ arbitrary hosts. But for our provider, this could be a major difference among other solutions if we want to support use case such as managing resources on remote virtual machines.